src/main/webapp/search-script.js

```diff
@@ -0,0 +1,122 @@
+  const profName = document.getElementById("search-prof").value;
+  const termName = document.getElementById("search-term").value;
+  const units = collectUnitsFromSearch();
+  const school = getUserSchool();
+  courseResults.innerHTML = "";
+  const url = new URL("/search", window.location.origin);
+  url.searchParams.set("courseName", courseName);
+  url.searchParams.set("profName", profName);
+  url.searchParams.set("units", units);
+  url.searchParams.set("term", termName);
+  url.searchParams.set("schoolName", school)
+  fetch(url)
+  .then(response => response.json())
+  .then((searchResults) => {
+    if (searchResults.hasOwnProperty("message")) { // TODO: Make a div for this message so specific styling is easier.
+        courseResults.innerHTML += searchResults.message + "<br />";
+    }
+    const courses = JSON.parse(searchResults.courses);
+    courses.forEach(course => courseResults.appendChild(createListElement(course)));
+  });
+}
+
+/* Creates an <li> element containing the course link and name. */
+function createListElement(course) {
+  const liElement = document.createElement('li');
+  const link = document.createElement('a');
+  const url = new URL("/course.html", window.location.origin);
+  url.searchParams.set("term-key", course.termKey);
+  url.searchParams.set("course-key", course.courseKey);
+  const school = getUserSchool();
+  url.searchParams.set("school-name", school);
+  link.setAttribute('href', url);
+  link.innerText = course.name + " - " + course.professor + " (" + course.term + ")" ;
+  liElement.appendChild(link);
+  return liElement;
+}
+
+function getUserSchool() {
+  const auth2 = gapi.auth2.getAuthInstance();
+  const profile = auth2.currentUser.get().getBasicProfile();
+  const email = profile.getEmail();
+  const start = email.indexOf('@');
+  const end = email.lastIndexOf('.');
+  const school = email.substring(start+1, end);
+  return school;
+}
+
+function addCourse() {
+  const courseName = document.getElementById("course-name").value;
+  const profName = document.getElementById("prof-name").value;
+  const termName = document.getElementById("term").value;
+  const units = collectUnitsFromInput();
+  const schoolName = getUserSchool();
+  const url = new URL("/search", window.location.origin);
+  url.searchParams.set("course-name", courseName);
+  url.searchParams.set("prof-name", profName);
+  url.searchParams.set("num-units", units);
+  url.searchParams.set("term", termName);
+  url.searchParams.set("school-name", schoolName);
+  fetch(url, {method:"POST"});
+}
+
+async function onSignIn(googleUser) {
+  const profile = googleUser.getBasicProfile();
+  const token = googleUser.getAuthResponse().id_token;
+  const url = new URL("/login", window.location.origin);
+  url.searchParams.set("token", token);
+  const response = await fetch(url, {method:"POST"});
+  const id = await response.json();
+    if (id.verified) { // Successful sign-in.
```

Comment:
Don't need indents.

If you `npm install -g prettier` in terminal, you should be able to run `prettier /path/to/this/file.js` to format it manually if the automatic thing isn't working.